pkg/cmd/clustertask/create_test.go

```diff
@@ -0,0 +1,237 @@
+				},
+			},
+		},
+	}
+
+	version := "v1alpha1"
+	tdc := testDynamic.Options{}
+	dynamic, err := tdc.Client(
+		cb.UnstructuredT(tasks[0], version),
+	)
+	if err != nil {
+		t.Errorf("unable to create dynamic client: %v", err)
+	}
+
+	cs, _ := test.SeedTestData(t, pipelinetest.Data{Tasks: tasks})
+	p := &test.Params{Tekton: cs.Pipeline, Kube: cs.Kube, Dynamic: dynamic}
+	p.SetNamespace("default")
+	cs.Pipeline.Resources = cb.APIResourceList(version, []string{"task"})
+
+	clustertask := Command(p)
+
+	out, err := test.ExecuteCommand(clustertask, "create", "ct", "--from-task", "t-1")
+	if err == nil {
+		t.Errorf("Expected error got nil")
+	}
+	expected := "Error: Task t-1 does not exist in namespace default\n"
+	test.AssertOutput(t, expected, out)
+}
+
+func TestCreate(t *testing.T) {
+	tasks := []*v1alpha1.Task{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "t-1",
+				Namespace: "default",
+			},
+			Spec: v1alpha1.TaskSpec{
+				TaskSpec: v1beta1.TaskSpec{
+					Description: "a task",
+				},
+			},
+		},
+	}
+
+	version := "v1alpha1"
+	tdc := testDynamic.Options{}
+	dynamic, err := tdc.Client(
+		cb.UnstructuredT(tasks[0], version),
+	)
+	if err != nil {
+		t.Errorf("unable to create dynamic client: %v", err)
+	}
+
+	cs, _ := test.SeedTestData(t, pipelinetest.Data{Tasks: tasks})
+	p := &test.Params{Tekton: cs.Pipeline, Kube: cs.Kube, Dynamic: dynamic}
+	p.SetNamespace("default")
+	cs.Pipeline.Resources = cb.APIResourceList(version, []string{"task", "clustertask"})
+
+	clustertask := Command(p)
+
+	out, err := test.ExecuteCommand(clustertask, "create", "ct-1", "--from-task", "t-1")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	expected := "ClusterTask ct-1 created from Task t-1 present in namespace default\n"
+	test.AssertOutput(t, expected, out)
+}
+
+func TestCreate_InNamespace(t *testing.T) {
+	tasks := []*v1alpha1.Task{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "t-1",
+				Namespace: "ns",
+			},
+			Spec: v1alpha1.TaskSpec{
+				TaskSpec: v1beta1.TaskSpec{
+					Description: "a task",
+				},
+			},
+		},
+	}
+
+	namespaces := []*corev1.Namespace{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ns",
+			},
+		},
+	}
+
+	version := "v1alpha1"
+	tdc := testDynamic.Options{}
+	dynamic, err := tdc.Client(
+		cb.UnstructuredT(tasks[0], version),
+	)
+	if err != nil {
+		t.Errorf("unable to create dynamic client: %v", err)
+	}
+
+	cs, _ := test.SeedTestData(t, pipelinetest.Data{Tasks: tasks, Namespaces: namespaces})
+	p := &test.Params{Tekton: cs.Pipeline, Kube: cs.Kube, Dynamic: dynamic}
+	cs.Pipeline.Resources = cb.APIResourceList(version, []string{"task", "clustertask"})
+
+	clustertask := Command(p)
+
+	out, err := test.ExecuteCommand(clustertask, "create", "--from-task", "t-1", "-n", "ns")
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	expected := "ClusterTask t-1 created from Task t-1 present in namespace ns\n"
+	test.AssertOutput(t, expected, out)
```

Comment:
Can we assert the output of clustertask yaml after creation in the form of golden file ??